helpers_test.go

```diff
@@ -17,14 +17,23 @@ func init() {
 	deep.CompareUnexportedFields = true
 }
 
-// nilErr tests for non-nil errors
-func nilErr(tb testing.TB, err error) {
+// assertErrorIsNil tests for non-nil errors
+func assertErrorIsNil(tb testing.TB, err error) {
 	tb.Helper()
 	if err != nil {
 		tb.Fatalf("\nunexpected error: %s", err.Error())
 	}
 }
 
+// assertKeyExists asserts that the provided key exists within the map
+func assertKeyExists(tb testing.TB, m map[string]interface{}, key string) {
+	tb.Helper()
+	_, exists := m[key]
```

Comment:
nit: maybe refactor to conform to Effective Go's ["comma ok"][commaok] idiom:

```go
tb.Helper()
if _, ok := m[key]; !ok {
  tb.Fatalf("key %s does not exist", key)
}
```

[commaok]: https://golang.org/doc/effective_go.html#maps

Reply:
I'm personally not a fan of that idiom in most cases

Reply:
I don't seek to be dogmatic about styling but I do feel we should favor consistency unless we have good reasons to deviate, and I do feel good reasons to present themselves from time to time. Is there an advantage or other reasoning to not use "comma ok" idiom here?

Reply:
I disagree that there is consistency in its usage:
```
~/go/src/github.com/hashicorp/vault $ ack "if ok " --ignore-dir vendor | wc -l
     236
~/go/src/github.com/hashicorp/vault $ ack "; ok \{" --ignore-dir vendor | wc -l
     328
```

The utilization of in-line if statements frequently leads to less readable code because it pushes the logic further right, breaking the [Return early rather than nesting deeply](https://dave.cheney.net/2020/02/23/the-zen-of-go#_return_early_rather_than_nesting_deeply) idiom from The Zen of Go. Taken to an extreme, it can lead to code like this:
```
if _, err := c.provider.UpdateApplicationPasswordCredentials(ctx, appObjID,
	graphrbac.PasswordCredentialsUpdateParameters{
		Value: &curCreds,
	},
); err != nil {
	if strings.Contains(err.Error(), "size of the object has exceeded its limit") {
		err = errors.New("maximum number of Application passwords reached")
	}
	return "", "", errwrap.Wrapf("error updating credentials: {{err}}", err)
}
```
https://github.com/hashicorp/vault-plugin-secrets-azure/blob/master/client.go#L197-L206

Having an extra line before an if statement is extremely cheap and aids readability because the structure is consistent with most other conditionals:
```
doThing()
if doThing failed {
  handle error
}
```

All that being said: because this is a simple use case I'll change it to an in-line if in order to move this PR forward.

Reply:
I appreciate the counter examples as much as I appreciate your willingness to compromise 😄